ui/src/track-layout/layout-design-api.ts

```diff
@@ -0,0 +1,32 @@
+import { TRACK_LAYOUT_URI } from 'track-layout/track-layout-api';
+import { getNonNull, postNonNull, putNonNull } from 'api/api-fetch';
+import { LayoutDesignId, TimeStamp } from 'common/common-model';
+import { asyncCache } from 'cache/cache';
+
+const designCache = asyncCache<string, LayoutDesign[]>();
+
+const baseUri = `${TRACK_LAYOUT_URI}/layout-design`;
+
+export type LayoutDesignSaveRequest = {
+    name: string;
+    estimatedCompletion: TimeStamp;
+    designState: 'ACTIVE' | 'DELETED' | 'COMPLETED';
```

Comment:
Pitäiskö tällä olla erillinen tyyppi vai riittääkö että on vain tälle kentälle määritellyt mahdolliset arvot?

Reply:
Toistaiseksi se riittää, joten en lähtenyt tekemään omaa tyyppiään tästä. Tällä arvollahan ei tehdä mitään tällä hetkellä, eikä välttämättä tullakaan tekemään frontissa oikeastaan mitään (designin asettamista koodissa `DELETED`- tai myöhemmin `COMPLETED`-tilaan lukuunottamatta,) joten tämä ei ainakaan toistaiseksi tarvitse esimerkiksi käännöstä. Sanoisin että pidetään tämä näin siihen saakka jos/kun joku kovempi tarve ilmaantuu, ja erotetaan tämä sitten omakseen